### PR TITLE
apiDoc /chat Documentation

### PR DESCRIPTION
**Pull request by librarianmage**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
#8087 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

Fix documentation for /chat API route

[//]: # (Put User ID in here - found in Settings -> API)

----
UUID:

### CODE REVIEW

File: website/server/controllers/api-v3/chat.js [hunk 1]
```diff
@@ -207,8 +207,8 @@ api.likeChat = {
  *
  * @apiUse GroupNotFound
  * @apiUse MessageNotFound
- * @apiError FlagOwnMessage Chat messages cannot be flagged by the author of the message
- * @apiError AlreadyFlagged Chat messages cannot be flagged more than once by a user
+ * @apiError (404) FlagOwnMessage Chat messages cannot be flagged by the author of the message
```
Comment (TheHollidayInn) [inline website/server/controllers/api-v3/chat.js]: Should this have the {NotFound} tag as well?

**Comment by TheHollidayInn** [comment]
Looks good! Just had one comment.

**Comment by librarianmage** [comment]
@TheHollidayInn Updated! Thank you for taking a look

File: website/server/controllers/api-v3/chat.js [hunk 2]
```diff
@@ -412,9 +412,9 @@ api.seenChat = {
  * @apiName DeleteChat
  * @apiGroup Chat
  *
- * @apiParam {String} previousMsg Query parameter - The last message fetched by the client so that the whole chat will be returned only if new messages have been posted in the meantime
- * @apiParam {UUID} groupId The group _id ('party' for the user party and 'habitrpg' for tavern are accepted)
- * @apiParam {UUID} chatId The chat message id
+ * @apiParam (Query) {String} previousMsg The last message fetched by the client so that the whole chat will be returned only if new messages have been posted in the meantime
```
Comment (TheHollidayInn) [inline website/server/controllers/api-v3/chat.js]: I'm pretty sure this needs to be a UUID like this: https://github.com/HabitRPG/habitica/pull/8277/files#diff-ade116c537945ae89e2f5f059d4d577fR80

**Comment by TheHollidayInn** [comment]
No problem! Thanks for working on this. I think the changes are good, but after reading the parent ticket 8087, I think I can review this a little more thoroughly. I see a few missing descriptions and errors un document for instance. I'll draft a list for each enpoint, if that works for you.

File: website/server/controllers/api-v3/chat.js [hunk 3]
```diff
@@ -207,8 +207,8 @@ api.likeChat = {
  *
  * @apiUse GroupNotFound
  * @apiUse MessageNotFound
- * @apiError FlagOwnMessage Chat messages cannot be flagged by the author of the message
- * @apiError AlreadyFlagged Chat messages cannot be flagged more than once by a user
+ * @apiError (404) {NotFound} FlagOwnMessage Chat messages cannot be flagged by the author of the message
+ * @apiError (404) {NotFound} AlreadyFlagged Chat messages cannot be flagged more than once by a user
```
Comment (Alys) [inline website/server/controllers/api-v3/chat.js:227]: FlagOwnMessage shouldn't exist. It is possible (deliberately) for the author to flag their own message.

**Comment by librarianmage** [comment]
@TheHollidayInn @Alys Updated!

**Comment by librarianmage** [comment]
Can this please be reviewed?

**Comment by TheHollidayInn** [comment]
I have this on my list to review today - sorry for the delay.

**Comment by librarianmage** [comment]
Not a problem

**Comment by TheHollidayInn** [comment]
Get Chat
 - Needs apiDescription
 - Needs apiError

Post Chat
 - Needs apiDescription
 - Needs apiError fro chat revoked (if you can make this a translated string that would also help)

Like Chat
- Needs apiDescription
- Needs apiError for messageGroupChatLikeOwnMessage

Flag Chat
- needs apiError for groupIdRequired, chatIdRequired, messageGroupChatFlagAlreadyReported

Clear Flag Chat
- needs apiError for groupIdRequired, chatIdRequired

Seen Chat
- needs apiDescription
- Needs apiError for groupIdRequired

Delete Chat
- Needs apiDescription
- previousMessage should say that this is a a chat ID. 
- needs apiError for groupIdRequired, chatIdRequired, onlyCreatorOrAdminCanDeleteChat

**Comment by librarianmage** [comment]
@TheHollidayInn Updated! Can you please review?

**Comment by TheHollidayInn** [comment]
Thanks! I will review this tomorrow night :D

**Comment by TheHollidayInn** [comment]
Noted towards your Tier progress! Thank you :D

**Pull request merged**